### PR TITLE
#162948994 User should be able to create articles

### PR DESCRIPTION
**Pull request by Hesbon5600**
**What does this PR do?**
 _A user should be able to:_
- Create an article
- Retrieve all articles
- Retrieve one article
- Edit an article he/she created
- Delete an article he/she created

**How should this be manually tested?**

1. Run the server:
`python manage.py runserver`

1. From postman:

    1. Create an article by sending a `post` request to the '/api/v1/articles/' end point with the data body 
    ```source-json
    {
        "article": {
        "title": "Test Article",
        "description": "Test description for the article",
        "body": "Test body for the article"
        }
    }
    ```
    2. Retrieve all articles by sending a `get` request to the  '/api/v1/articles/' end point
    3. Retrieve a single article by sending a `get` request to the  '/api/v1/articles/`<slug>`/' end point and pass the articles's slug in the url
    4. Update an  article by sending a `put` request to the  '/api/v1/articles/`<slug>`/' end point and pass the articles's slug in the url. Pass the following as the body
    ``` source-json
    {
    "article": {
        "title": "Updated title",
        "description": "Updated description",
        "body": "Updated body"
    }
    }
    ```
    5. Delete an  article by sending a `delete` request to the  '/api/v1/articles/`<slug>`/' end point

**Any background context you want to provide?**  
None

**What are the relevant pivotal tracker stories?**
[#162948994](https://www.pivotaltracker.com/story/show/162948994)

**Screenshots**
   1. Create an article
screenshot 2019-01-22 at 12 34 09
   2. Retrieve all articles
screenshot 2019-01-22 at 12 35 50
   3. Retrieve one article
screenshot 2019-01-22 at 12 36 42
   4. Update an article
screenshot 2019-01-22 at 12 39 17
   5. Delete an article
screenshot 2019-01-22 at 12 40 37

### CODE REVIEW

**Review by Nyakaru — CHANGES_REQUESTED** [review]
@Hesbon5600 Kindly squash your commits into one

**Review by kzyangiro — CHANGES_REQUESTED** [review]
You need to have only one migration file. @Hesbon5600

File: authors/apps/articles/tests/test_articles.py [hunk 1]
```diff
@@ -0,0 +1,131 @@
...
+        url = self.single_article_url(slug)
+        response = self.client.delete(
+            url, format='json', HTTP_AUTHORIZATION='Token ' +
+            self.login_user())
+        message = 'Article Deleted Successfully'
+        self.http_200_ok(response)
+        self.base_articles(message, response)
+        message = "Article Deleted Successfully"
+        self.base_articles(message, response)
+
```
Comment (kzyangiro) [inline authors/apps/articles/tests/test_articles.py:119]: Duplicate here @Hesbon5600

File: authors/apps/articles/views.py [hunk 2]
```diff
@@ -0,0 +1,84 @@
+from rest_framework.generics import (
+    ListCreateAPIView, RetrieveUpdateDestroyAPIView)
+from rest_framework.permissions import IsAuthenticatedOrReadOnly
+from rest_framework.response import Response
+from rest_framework import status
+from rest_framework.exceptions import ValidationError
+# from django.contrib.auth.models import AnonymousUser
```
Comment (kzyangiro) [inline authors/apps/articles/views.py]: Remove unnecessary comments

**Review by Nyakaru — APPROVED** [review]

File: authors/apps/articles/views.py (hunk 2 above)
Comment (kzyangiro) [inline authors/apps/articles/views.py]: You are nolonger using this comment, kindly remove it. @Hesbon5600

**Review by JoshuaKodhe — APPROVED** [review]
The update image url has been fixed and work as should

**Review by kzyangiro — APPROVED** [review]

**Pull request merged**